plugins/registry.ts

```diff
@@ -187,20 +187,19 @@ export default class PluginRegistry {
     // Add a button to the channel intro message.
     // Accepts the following:
     // - icon - React element to use as the button's icon
-    // - action - a function called when the button is clicked, passed the channel and channel member as arguments
-    // - tooltipText - string or React element shown for tooltip appear on hover
+    // - action - a function called when the button is clicked
+    // - text - a localized string to use as the button's text
     registerChannelIntroButtonAction = reArg([
         'icon',
         'action',
```

Comment:
Without this, `'Boards'` in https://github.com/mattermost/focalboard/blob/eceb406535b80cd645678b9273c1ace5c62f689e/mattermost-plugin/webapp/src/index.tsx#L332 won't get mapped to `text` in the destructured param. (Unless it was switched to the object arg signature.)

```suggestion
        'action',
        'text',
```

Reply:
Yep, that's expected. The text doesn't match ("Create a Board" vs "Boards"), and I didn't want to have to break that, so I just chose not to add `text` as an option if you're using the old API.

Personally, I'd be a fan of deprecating the non-object way of calling these at some point as well, so I don't think it's as important for that to support all the options. I haven't actually ran that by anyone yet though

Reply:
Actually, we need to keep 'text' in the mapping for the time being. `reArg` has a keys check to verify the mapping is supposed to be done or not https://github.com/mattermost/mattermost-webapp/blob/d2e60742b5cbd342981e542cbeb31eb802063d6e/utils/func.ts#L30


Without consistent keys between `keyOrder` and `TArgs`, when calling via the new object-arg signature, `reArg` treats the object provided as `icon` (first argument), and this causes a hard crash.


I'm all for the deprecating route, but `reArg` will have to be improved a bit before we can do hard cutoffs like this.

Reply:
Yeah, I've wanted to do something similar, but with a new package instead of adding it to `@mattermost/types`. I've been preferring the idea of more smaller packages lately that are more focused on use case rather than "here's all  this sort of code" like how most of our code base is organized now. There's definitely a potential that that just means everyone needs to add 12 `@mattermost/` dependencies at the start of a project that I am a bit worried about though.

Good catch. I'll re-add the `text` parameter but mark that it's unused then